src/internal/connector/graph/errors.go

```diff
@@ -79,7 +80,7 @@ func IsErrInvalidDelta(err error) bool {
 		return true
 	}
 
-	if hasErrorCode(err, errCodeSyncStateNotFound, errCodeResyncRequired) {
+	if hasErrorCode(err, errCodeSyncStateNotFound, errCodeResyncRequired, errCodeResyncRequired2) {
```

Comment:
We could maybe lowercase and compare it inside `hasErrorCode`, but I was not sure if we wanted to take that route.